contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,451 @@
+            msg.event_info.per_address_limit.to_string(),
+        )
+        .add_attribute("base_poap_uri", &msg.event_info.base_poap_uri)
+        .add_attribute("event_uri", &msg.event_info.event_uri)
+        .add_attribute("cw721_code_id", &msg.cw721_code_id.to_string())
+        .add_submessages(sub_msgs))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::EnableMint {} => execute_set_mint_enabled(deps, info, true),
+        ExecuteMsg::DisableMint {} => execute_set_mint_enabled(deps, info, false),
+        ExecuteMsg::Mint {} => {
+            let recipient_addr = info.sender.clone();
+            execute_mint(deps, env, info, ACTION_MINT, recipient_addr, false, false)
+        }
+        ExecuteMsg::MintTo { recipient } => {
+            let recipient_addr = deps.api.addr_validate(&recipient)?;
+            execute_mint(deps, env, info, ACTION_MINT_TO, recipient_addr, true, true)
+        }
+        ExecuteMsg::UpdateEventInfo {
+            start_time,
+            end_time,
+        } => execute_update_event_info(deps, env, info, start_time, end_time),
+        ExecuteMsg::UpdateAdmin { new_admin } => execute_update_admin(deps, info, new_admin),
+        ExecuteMsg::UpdateMinter { new_minter } => execute_update_minter(deps, info, new_minter),
+    }
+}
+
+fn execute_set_mint_enabled(
+    deps: DepsMut,
+    info: MessageInfo,
+    mint_enabled: bool,
+) -> Result<Response, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    // Check that the sender is the admin
+    if info.sender != config.admin {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    config.mint_enabled = mint_enabled;
+    // Save the new configurations
+    CONFIG.save(deps.storage, &config)?;
+
+    let action = if mint_enabled {
+        ACTION_ENABLE_MINT
+    } else {
+        ACTION_DISABLE_MINT
+    };
+
+    Ok(Response::new()
+        .add_attribute("action", action)
+        .add_attribute("sender", info.sender))
+}
+
+fn execute_mint(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    action: &str,
+    recipient_addr: Addr,
+    bypass_mint_enable: bool,
+    check_authorized_to_mint: bool,
+) -> Result<Response, ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let event_info = EVENT_INFO.load(deps.storage)?;
+
+    // Check if the event is started
+    if !event_info.is_started(&env.block.time) {
+        return Err(ContractError::EventNotStarted {
+            current_time: env.block.time,
+            start_time: event_info.start_time,
+        });
+    }
+
+    // Check if the event is ended
+    if event_info.is_ended(&env.block.time) {
+        return Err(ContractError::EventTerminated {
+            current_time: env.block.time,
+            end_time: event_info.end_time,
+        });
+    }
```

Comment:
Can we merge them into one method `event_info.validate_date()`?

Reply:
With this we will move some contract logic inside `EventInfo` i do prefer keeping the responsibilities separated between them